overlord/servicestate/internal/quotas.go

```diff
@@ -103,13 +103,22 @@ func PatchQuotas(st *state.State, grps ...*quota.Group) (map[string]*quota.Group
 		return nil, fmt.Errorf("cannot update quota%s %s: %v", plural, updated, err)
 	}
 
+	// Verify that the update of the new quota groups will result in
+	// correct nesting of groups. Execute this verification after updating
+	// group pointers in ResolveCrossReferences.
+	for _, grp := range grps {
+		if err := grp.VerifyNesting(); err != nil {
```

Comment:
see my comment elsewhere about the name and whether it checks enough for this

Reply:
Thanks for pointing this out!